plugins/catalog/src/alpha/components/EntityTabs/EntityTabsGroup.tsx

```diff
@@ -138,22 +142,41 @@ const styles = (theme: Theme) =>
 
 type EntityTabsGroupItem = {
   id: string;
-  index: number;
   label: string;
   path: string;
-  group: string;
+  icon?: string | ReactElement;
 };
 
 type EntityTabsGroupProps = TabProps & {
   classes?: Partial<ReturnType<typeof styles>>;
   indicator?: ReactNode;
-  highlightedButton?: number;
+  highlightedButton?: string;
   items: EntityTabsGroupItem[];
-  onSelectTab: MouseEventHandler<HTMLAnchorElement>;
+  onSelectTab?: MouseEventHandler<HTMLAnchorElement>;
+  showIcons?: boolean;
 };
 
+function resolveIcon(
+  icon: string | ReactElement | undefined,
+  iconsApi: IconsApi,
+  showIcons: boolean,
+) {
+  if (!showIcons) {
+    return undefined;
+  }
+  const itemIcon = icon;
+  if (typeof itemIcon === 'string') {
+    const Icon = iconsApi.getIcon(itemIcon);
+    if (Icon) {
+      return <Icon />;
+    }
```

Comment:
Should this one have an `else { return undefined; }`? I thought this function was meant to just return `JSX.Element | undefined`, not `string`, which would happen on the line below otherwise